controlplane/pkg/nsm/nsm_heal.go

```diff
@@ -184,20 +240,34 @@ func (srv *networkServiceManager) waitRemoteNSE(ctx context.Context, clientConne
 	st := time.Now()
 
 	nseRequest := &registry.FindNetworkServiceRequest{
-		NetworkServiceName: clientConnection.GetNetworkService(),
+		NetworkServiceName: networkService,
 	}
 
 	defer func() {
-		logrus.Infof("Complete Waiting for Remote NSE/NSMD with network service %s. Since elapsed: %v", clientConnection.Endpoint.NetworkService.Name, time.Since(st))
+		logrus.Infof("Complete Waiting for Remote NSE/NSMD with network service %s. Since elapsed: %v", networkService, time.Since(st))
 	}()
 
 	for {
-		logrus.Infof("NSM: RemoteNSE:  Waiting for NSE with network service %s. Since elapsed: %v", clientConnection.Endpoint.NetworkService.Name, time.Since(st))
+		logrus.Infof("NSM: RemoteNSE:  Waiting for NSE with network service %s. Since elapsed: %v", networkService, time.Since(st))
 
 		endpointResponse, err := discoveryClient.FindNetworkService(ctx, nseRequest)
 		if err == nil {
 			for _, ep := range endpointResponse.NetworkServiceEndpoints {
-				if ep.EndpointName == clientConnection.Endpoint.NetworkserviceEndpoint.EndpointName {
+				if ignoreEndpoint != "" && ep.EndpointName == ignoreEndpoint {
+					// Skip ignored endpoint
+					continue
+				}
+				if srv.getNetworkServiceManagerName() == ep.GetNetworkServiceManagerName() {
+					// Another local endpoint is found, success.
+					reg := &registry.NSERegistration{
+						NetworkServiceManager:  endpointResponse.GetNetworkServiceManagers()[ep.GetNetworkServiceManagerName()],
+						NetworkserviceEndpoint: ep,
+						NetworkService:         endpointResponse.GetNetworkService(),
+					}
+					clientConnection.Endpoint = reg
+					return true
+				}
+				if clientConnection.Endpoint != nil && ignoreEndpoint == "" && ep.EndpointName == clientConnection.Endpoint.NetworkserviceEndpoint.EndpointName {
```

Comment:
Maybe I mistaken but seems like if `ignoreEndpoint != ""` we have no chance to choose remote endpoint

Reply:
Changed.